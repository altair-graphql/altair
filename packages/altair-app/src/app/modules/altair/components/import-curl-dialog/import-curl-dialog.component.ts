import { Component, Input, Output, EventEmitter } from '@angular/core';

@Component({
  selector: 'app-import-curl-dialog',
  templateUrl: './import-curl-dialog.component.html',
  styleUrls: ['./import-curl-dialog.component.scss'],
})
export class ImportCurlDialogComponent {
  @Input() showImportCurlDialog = false;
  @Output() toggleDialogChange = new EventEmitter();
  @Output() importCurlChange = new EventEmitter<string>();

  textAreaInput = '';

  constructor() {}

  importInput() {
    this.toggleDialogChange.next(false);
    this.importCurlChange.next(this.textAreaInput);
  }
}
