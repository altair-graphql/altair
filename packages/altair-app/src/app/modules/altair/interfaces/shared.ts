export interface IDictionary<V = any> {
  [key: string]: V;
}

export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

export type TODO = any;

export type UnknownError = TODO;

export interface TrackByIdItem {
  id: string;
}
