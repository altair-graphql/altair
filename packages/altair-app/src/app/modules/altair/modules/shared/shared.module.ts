import { NgModule, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';

import { DragDropModule } from '@angular/cdk/drag-drop';
import { TranslateModule } from '@ngx-translate/core';
import { NgPipesModule } from 'ngx-pipes';
import { NguiAutoCompleteModule } from '@ngui/auto-complete';
import { MarkdownModule } from 'ngx-markdown';
import { HotToastModule } from '@ngneat/hot-toast';
import { ResizableModule } from 'angular-resizable-element';

import { NzButtonModule } from 'ng-zorro-antd/button';
import { NzFormModule } from 'ng-zorro-antd/form';
import { NzSelectModule } from 'ng-zorro-antd/select';
import { NzModalModule } from 'ng-zorro-antd/modal';
import { NzDropDownModule } from 'ng-zorro-antd/dropdown';
import { NzSwitchModule } from 'ng-zorro-antd/switch';
import { NzListModule } from 'ng-zorro-antd/list';
import { NzRadioModule } from 'ng-zorro-antd/radio';
import { NzAlertModule } from 'ng-zorro-antd/alert';
import { NzLayoutModule } from 'ng-zorro-antd/layout';
import { NzMenuModule } from 'ng-zorro-antd/menu';
import { NzCollapseModule } from 'ng-zorro-antd/collapse';
import { NzInputModule } from 'ng-zorro-antd/input';
import { NzInputNumberModule } from 'ng-zorro-antd/input-number';
import { NzAutocompleteModule } from 'ng-zorro-antd/auto-complete';
import { NzCheckboxModule } from 'ng-zorro-antd/checkbox';
import { NzToolTipModule } from 'ng-zorro-antd/tooltip';
import { NzTabsModule } from 'ng-zorro-antd/tabs';
import { NzTableModule } from 'ng-zorro-antd/table';
import { NzTreeSelectModule } from 'ng-zorro-antd/tree-select';
import { NzGridModule } from 'ng-zorro-antd/grid';
import { NzPopoverModule } from 'ng-zorro-antd/popover';

import { IconsModule } from '../icons/icons.module';
import { NzAvatarModule } from 'ng-zorro-antd/avatar';

const AntdComponentModules = [
  NzButtonModule,
  NzInputModule,
  NzInputNumberModule,
  NzAutocompleteModule,
  NzCheckboxModule,
  NzFormModule,
  NzGridModule,
  NzTabsModule,
  NzSelectModule,
  NzModalModule,
  NzDropDownModule,
  NzSwitchModule,
  NzListModule,
  NzRadioModule,
  NzAlertModule,
  NzLayoutModule,
  NzMenuModule,
  NzCollapseModule,
  NzToolTipModule,
  NzTableModule,
  NzTreeSelectModule,
  NzPopoverModule,
  NzAvatarModule,
  // NgZorroAntdModule,
];

@NgModule({
  imports: [
    CommonModule,
    DragDropModule,
    TranslateModule.forRoot(),
    NgPipesModule,
    NguiAutoCompleteModule, // use antd
    MarkdownModule.forRoot(),
    ResizableModule,
    IconsModule,
    // ToastrModule.forRoot({
    //   newestOnTop: false,
    //   closeButton: true,
    //   positionClass: 'toast-top-center',
    //   enableHtml: true,
    //   countDuplicates: true,
    //   preventDuplicates: true,
    //   resetTimeoutOnDuplicate: true,
    // }),
    // HotToastModule.forRoot({
    //   position: 'top-center',
    //   reverseOrder: true,
    //   dismissible: true,
    //   closeStyle: {
    //     backgroundImage: `none`,
    //   },
    // }),
    ...AntdComponentModules,
  ],

  // DON'T FORGET TO EXPORT MODULE
  exports: [
    DragDropModule,
    TranslateModule,
    NgPipesModule,
    NguiAutoCompleteModule,
    MarkdownModule,
    ResizableModule,
    IconsModule,
    HotToastModule,
    ...AntdComponentModules,
  ],
})
export class SharedModule {
  static forRoot(): ModuleWithProviders<SharedModule> {
    return {
      ngModule: SharedModule,
      providers: [],
    };
  }
}
