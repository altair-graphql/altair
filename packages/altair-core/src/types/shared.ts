export type IDictionary<V = any> = Record<string, V>;

export type TODO = any;

// export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

export interface BaseDocument {
  id: string;
  created_at?: number;
  updated_at?: number;
}

export interface BaseOwnableDocument extends BaseDocument {
  ownerUid: string;
}

type TypedOmit<T, K extends keyof T> = Omit<T, K>;
export type CreateDTO<
  T extends BaseDocument | BaseOwnableDocument
> = T extends BaseOwnableDocument
  ? TypedOmit<T, keyof BaseOwnableDocument>
  : TypedOmit<T, keyof BaseDocument>;
export type UpdateDTO<T extends BaseDocument | BaseOwnableDocument> = CreateDTO<
  T
> &
  Pick<T, 'id'>;
