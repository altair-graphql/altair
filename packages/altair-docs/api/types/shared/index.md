# types/shared

## Index

| Member | Description |
| :------ | :------ |
| [BaseDocument](interfaces/BaseDocument.md) | - |
| [BaseOwnableDocument](interfaces/BaseOwnableDocument.md) | - |
| [CreateDTO](type-aliases/CreateDTO.md) | - |
| [IDictionary](type-aliases/IDictionary.md) | - |
| [NoExtraProperties](type-aliases/NoExtraProperties.md) | - |
| [TODO](type-aliases/TODO.md) | - |
| [UpdateDTO](type-aliases/UpdateDTO.md) | - |
