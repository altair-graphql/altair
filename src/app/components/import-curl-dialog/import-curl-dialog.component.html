<nz-modal
  [nzVisible]="showImportCurlDialog"
  (nzVisibleChange)="toggleDialogChange.next($event)"
  [nzTitle]="modalTitle"
  [nzContent]="modalContent"
  [nzFooter]="modalFooter"
  (nzOnCancel)="toggleDialogChange.next($event)"
>
  <ng-template #modalTitle>
    <div class="app-dialog-header">
      <div class="app-dialog-title">{{ 'IMPORT_CURL_TITLE_TEXT' | translate }}</div>
      {{ 'IMPORT_CURL_TITLE_SUBTEXT' | translate }}
    </div>
  </ng-template>

  <ng-template #modalContent>
    <div class="app-dialog-body">
      <div class="form-group">
        <textarea class="dialog-textarea" aria-label="text input" cols="30" rows="10" [(ngModel)]="textAreaInput"></textarea>
      </div>
    </div>
  </ng-template>

  <ng-template #modalFooter>
    <div class="app-dialog-footer">
      <button class="app-button active-primary right" track-id="close_import_from_curl_dialog" (click)="importInput()">{{ 'IMPORT_BUTTON' | translate }}</button>
    </div>
  </ng-template>
</nz-modal>
