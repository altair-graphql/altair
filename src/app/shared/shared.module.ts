import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { TranslateModule } from '@ngx-translate/core';
import { NgxElectronModule } from 'ngx-electron';
import { SortablejsModule } from 'angular-sortablejs';
import { NgPipesModule } from 'ngx-pipes';
import { ContextMenuModule } from 'ngx-contextmenu';
import { DndModule } from '@beyerleinf/ngx-dnd';
import { NguiAutoCompleteModule } from '@ngui/auto-complete';


@NgModule({
  imports: [
    CommonModule,
    TranslateModule.forRoot(),
    NgxElectronModule,
    SortablejsModule,
    NgPipesModule,
    NguiAutoCompleteModule,
    ContextMenuModule.forRoot(),
    DndModule.forRoot(),
  ],
  exports: [
    TranslateModule,
    NgxElectronModule,
    SortablejsModule,
    NgPipesModule,
    ContextMenuModule,
    DndModule,
    NguiAutoCompleteModule,
  ]
})
export class SharedModule { }
