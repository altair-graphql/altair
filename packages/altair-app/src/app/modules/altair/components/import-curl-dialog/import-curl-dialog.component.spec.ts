import { expect, describe, it } from '@jest/globals';

import { ImportCurlDialogComponent } from './import-curl-dialog.component';

import { SharedModule } from '../../modules/shared/shared.module';
import { NgxTestWrapper } from '../../../../../testing/wrapper';
import { mount } from '../../../../../testing';
import { MockModule } from 'ng-mocks';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('ImportCurlDialogComponent', () => {
  let wrapper: NgxTestWrapper<ImportCurlDialogComponent>;

  beforeEach(async () => {
    wrapper = await mount({
      component: ImportCurlDialogComponent,
      imports: [MockModule(SharedModule)],
      schemas: [NO_ERRORS_SCHEMA],
    });
  });

  it('should create', () => {
    expect(wrapper.componentInstance).toBeTruthy();
  });

  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
