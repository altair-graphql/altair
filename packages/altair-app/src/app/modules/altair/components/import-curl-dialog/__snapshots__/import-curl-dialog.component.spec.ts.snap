// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportCurlDialogComponent should render correctly 1`] = `
<app-import-curl-dialog>
  <nz-modal
    nzcloseicon="close-circle"
  >
    
    
    
    
    
    
  </nz-modal>
</app-import-curl-dialog>
`;
